### PR TITLE
Add user current location

### PR DESCRIPTION
**Pull request by Jonathan204**
Prompts the user to use their current location.

Displays a little pin at the users current location.

### CODE REVIEW

File: client/src/components/Map/Map.js [hunk 1]
```diff
@@ -11,20 +12,36 @@ class Map extends Component {
...
           defaultCenter={this.props.center}
+          center={this.state.center}
           defaultZoom={this.props.zoom}
         >
-          <AnyReactComponent
-            lat={49.895138}
-            lng={-97.138374}
-            text="My Marker"
+          <CurrLocation
+            lat={this.state.center.lat}
```
Comment (chris-mega) [inline client/src/components/Map/Map.js:43]: If the user doesn't allow location, the marker will be placed always in the center of the map (even if maps is dragged). Are we going to still draw a marker in this situation?

File: client/src/components/Map/Map.js [hunk 2]
```diff
@@ -11,20 +12,36 @@ class Map extends Component {
...
+    const latitude = position.coords.latitude
+    const longitude = position.coords.longitude
+    this.setState({
+      center: {lat: latitude, lng: longitude}
+    })
+  }
+
   render() {
     return (
       // Important! Always set the container height explicitly
```
Comment (chris-mega) [inline client/src/components/Map/Map.js:34]: Maybe we should do destructuring from state so we don't have to repeat `this.state...`
Something like `const { center } = this.state;`

**Review by chris-mega — COMMENTED** [review]
Looks good. Just some suggestions

File: client/src/components/Map/Map.js (hunk 1 above)
Comment (Jonathan204) [reply client/src/components/Map/Map.js:43]: I will make it so the marker only appears if the user allows his locations.

**Pull request merged**